tables/attributeset.py

```diff
@@ -330,8 +330,12 @@ def __getattr__(self, name):
             except UnicodeDecodeError:
                 # Object maybe pickled on python 2 and unpickled on python 3.
                 # encoding='bytes' was added in python 3.4 to resolve this.
+                # However 'bytes' mangles class attributes as they are
+                # unplicked as bytestrings. Hence try 'latin1' first.
                 # Ref: http://bugs.python.org/issue6784
                 try:
+                    retval = cPickle.loads(value, encoding='latin1')
+                except TypeError:
                     retval = cPickle.loads(value, encoding='bytes')
```

Comment:
@avalentino: Yes, indeed! This is starting to get ridiculous, but an extra try/except is needed. 502c40c